### ISSUE TITLE
error: could not build Objective-C module 'Apptimize'

### ISSUE THREAD

**Issue by Edudjr** [urbanairship/apptimize-ios-kit#1]
Hello,

I'm facing an issue while trying to use Apptimize inside a swift library/framework. It says it cannot build the objective-c module because of the ios 11 verification in `Apptimize.h`, even though the library is set to use v11.

The major problem is that including Apptimize in my project breaks the debugging for my internal framework which relies on Apptimize. Removing Apptimize from the project brings back the lldb debugger again.

This is the error:

```
warning: Swift error in scratch context: <module-includes>:1:9: note: in file included from <module-includes>:1:
#import "Headers/Apptimize.h"
        ^

error: /Users/eduardo.domene/Library/Developer/Xcode/DerivedData/ExampleLib-dfcguaonsvqexrfkaukjftpvwqjw/Build/Products/Debug-iphonesimulator/Apptimize.framework/Headers/Apptimize.h:16:2: error: "The Apptimize library uses features only available in iOS SDK 11.0 and later."
#error "The Apptimize library uses features only available in iOS SDK 11.0 and later."
 ^

error: could not build Objective-C module 'Apptimize'
```

And my Package.swift is:

```
// swift-tools-version:5.3
// The swift-tools-version declares the minimum version of Swift required to build this package.

import PackageDescription

let package = Package(
    name: "ExampleLib",
    platforms: [.iOS(.v11)],
    products: [
        .library(
            name: "ExampleLib",
            targets: ["ExampleLib"]),
    ],
    dependencies: [
        .package(name: "Apptimize",
                 url: "https://github.com/urbanairship/apptimize-ios-kit",
                 .exact("3.4.21")),
    ],
    targets: [
        .target(
            name: "ExampleLib",
            dependencies: ["Apptimize"]),
        .testTarget(
            name: "ExampleLibTests",
            dependencies: ["ExampleLib"]),
    ]
)

```

The issue is easy to reproduce:

**Step 1. Create a SPM package** 
- mkdir ExampleLib
- cd ExampleLib
- swift package init

**Step 2. Add Apptimize as a dependency**

**Step 3. Import Apptimize in a source code from the lib. ex:**
```
import Apptimize

struct ExampleLib {
    var text = "Hello, World!"
    
    public func doSomething(param: String) {
        print(param)
    }
}
```

**Step 4. Add a breakpoint in the lib function, and run a test for that function:**
```
final class ExampleLibTests: XCTestCase {
    func testExample() {
        ExampleLib().doSomething(param: "hello")
    }

    static var allTests = [
        ("testExample", testExample),
    ]
}
```

Step 5. Check the log:
```
warning: Swift error in scratch context: <module-includes>:1:9: note: in file included from <module-includes>:1:
#import "Headers/Apptimize.h"
        ^

error: /Users/eduardo.domene/Library/Developer/Xcode/DerivedData/ExampleLib-dfcguaonsvqexrfkaukjftpvwqjw/Build/Products/Debug-iphonesimulator/Apptimize.framework/Headers/Apptimize.h:16:2: error: "The Apptimize library uses features only available in iOS SDK 11.0 and later."
#error "The Apptimize library uses features only available in iOS SDK 11.0 and later."
 ^

error: could not build Objective-C module 'Apptimize'
```

ps: you can also get the same error again when trying to do `po param` in the breakpoint

example project:
https://github.com/Edudjr/ApptimizeExampleLib

printscreen:

<img width="1346" alt="Screenshot 2021-04-25 at 13 33 27" src="https://user-images.githubusercontent.com/6642478/115991884-e7734480-a5ca-11eb-8146-5d44b9645716.png">

ps: this is also reproducible using cocoapods.

**Comment by Edudjr** [comment]
I don't know why `Apptimize.h` doesn't recognize the __IPHONE_OS_VERSION_MIN_REQUIRED as the one we set in the Package. However, since the package won't build at all if I define a version lesser than v11 (like `platforms: [.iOS(.v10)]`), maybe we could remove or add a warning instead of an error in that min version verification at `Apptimize.h`?. When I do that, my package works fine and I can debug properly again.

**Comment by nickoto** [comment]
Thanks for the report @Edudjr; I will look at changing this asap.

**Comment by nickoto** [comment]
This should be fixed as of 3.4.25
